include/CBawFile.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018 Krzysztof "Criss" Szenk
+// This file is part of the "Irrlicht Engine" and "Build A World".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+// and on http://irrlicht.sourceforge.net/forum/viewtopic.php?f=2&t=49672
+
+#include "CBawFile.h"
+
+#include "ISkinnedMesh.h"
+#include "SSkinMeshBuffer.h"
+#include "CFinalBoneHierarchy.h"
+
+//! for C++11
+//using namespace std;
+
+
+namespace irr { namespace core
+{
+
+		size_t VariableSizeBlob<scene::ICPUMesh>::calcBlobSizeForObj(scene::ICPUMesh* _obj)
+		{
+			return sizeof(MeshBlobV1) + (_obj->getMeshBufferCount()-1) * sizeof(uint64_t);
+		}
+		void* VariableSizeBlob<scene::ICPUMesh>::allocMemForBlob(scene::ICPUMesh* _obj) { return malloc(calcBlobSizeForObj(_obj)); }
```

Comment:
does it not compile if you leave it unspecialized?

Reply:
I don't understand how I could leave it unspecialized

Reply:
"curiously recurring template" will allow us to declare static calcBlobSizeForObj and allocMemForBlob functions which can be "overriden" like virtual functions